File: leetcode/3019/number_of_changing_keys.cpp
#include <iostream>
#include <string>

using namespace std;

class Solution {

public:
  int countKeyChanges(string s) {
    int count = 0;
    char prev_char = tolower(s[0]);

    for (char curr_char : s) {
      curr_char = tolower(curr_char);
      if (curr_char != prev_char) {
        prev_char = curr_char;
        count++;
      };
    };

    return count;
  };
};

int main() {
}

File: leetcode/3019/number_of_changing_keys.py
class Solution:
    def countKeyChanges(self, s: str) -> int:
        s = s.lower()
        prev_c = s[0]
        count = 0
        for c in s[1:]:
            if not c == prev_c:
                prev_c = c
                count += 1

        return count
